src/main/java/org/openrewrite/java/migrate/lombok/LombokValueToRecord.java

```diff
@@ -0,0 +1,349 @@
+
+            if (fluentMethodName.length() == 0) {
+                return "";
+            }
+
+            final char firstMemberChar = fluentMethodName.charAt(0);
+            fluentMethodName.setCharAt(0, Character.toLowerCase(firstMemberChar));
+
+            return fluentMethodName.toString();
+        }
+
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(final J.ClassDeclaration cd, final ExecutionContext ctx) {
+            J.ClassDeclaration classDeclaration = super.visitClassDeclaration(cd, ctx);
+
+            if (!isRelevantClass(classDeclaration)) {
+                return classDeclaration;
+            }
+
+            final List<J.VariableDeclarations> memberVariables = findAllClassFields(classDeclaration);
+            if (hasMemberVariableAssignments(memberVariables)) {
+                return classDeclaration;
+            }
+
+            final List<Statement> bodyStatements = new ArrayList<>(classDeclaration.getBody().getStatements());
+            bodyStatements.removeAll(memberVariables);
+
+            classDeclaration = removeValueAnnotation(classDeclaration, ctx);
+            classDeclaration = classDeclaration
+                    .withKind(J.ClassDeclaration.Kind.Type.Record)
+                    .withType(buildRecordType(classDeclaration))
+                    .withBody(classDeclaration.getBody()
+                            .withStatements(bodyStatements)
+                    )
+                    .withPrimaryConstructor(mapToConstructorArguments(memberVariables));
+
+            if (useExactToString) {
+                classDeclaration = addExactToStringMethod(classDeclaration, memberVariables);
+            }
+
+            addToRecordTypeState(classDeclaration, memberVariables);
+
+            return maybeAutoFormat(cd, classDeclaration, ctx);
+        }
+
+        private J.ClassDeclaration addExactToStringMethod(final J.ClassDeclaration classDeclaration,
+                                                          final List<J.VariableDeclarations> memberVariables) {
+            return classDeclaration.withBody(TO_STRING_TEMPLATE
+                    .apply(new Cursor(getCursor(), classDeclaration.getBody()),
+                            classDeclaration.getBody().getCoordinates().lastStatement(),
+                            classDeclaration.getSimpleName(),
+                            memberVariablesToString(getMemberVariableNames(memberVariables))));
+        }
+
+        private static boolean isRelevantClass(final J.ClassDeclaration classDeclaration) {
+            return !isRecord(classDeclaration)
+                    && hasOnlyLombokValueAnnotation(classDeclaration)
+                    && !hasGenericTypeParameter(classDeclaration)
+                    && !hasExplicitMethods(classDeclaration)
+                    && !hasExplicitConstructor(classDeclaration)
+                    && classDeclaration.getType() != null;
```

Comment:
Maybe good to pull this last one up, as it's cheaper to check than some of the others.